engine/src/main/java/org/destinationsol/game/asteroid/FarAsteroid.java

```diff
@@ -49,7 +52,41 @@ public boolean shouldBeRemoved(SolGame game) {
 
     @Override
     public SolObject toObj(SolGame game) {
-        return game.getAsteroidBuilder().build(game, myPos, myTex, mySz, myAngle, myRotSpd, mySpd, myRemoveController);
+        Vector2 position = adjustDesiredPosition(game, myPos, mySz);
+        return game.getAsteroidBuilder().build(game, position, myTex, mySz, myAngle, myRotSpd, mySpd, myRemoveController);
+    }
+
+    /**
+     * Ensures that the position of any Asteroids do not overlap
+     * @param game SolGame instance
+     * @param desiredPosition Vector2 position that you would like the Asteroid to spawn at
+     * @param size float size of the asteroid
+     * @return the new Vector2 position of the asteroid, or desired if it does not overlap
+     */
+    private static Vector2 adjustDesiredPosition(SolGame game, Vector2 desiredPosition, float size) {
+        Vector2 newPosition = desiredPosition;
```

Comment:
Here, `newPosition` is a reference to desiredPosition. That means any change to one reflects in the other. Maybe you meant to do `newPosition = new Vector2(desiredPosition);`?
Either way, I recommend not creating a new vector here, and instead modifying desiredPosition instead (and hence make the return type void).

Also, since this method is already part of the class, why pass size and position as arguments? Simply make the method non-static, and take the variables from the class.

So, after all my suggested changes, this method becomes `private void adjustDesiredPosition(SolGame game)`